subsys/net/lib/azure_iot_hub/src/azure_iot_hub.c

```diff
@@ -32,17 +33,23 @@ LOG_MODULE_REGISTER(azure_iot_hub, CONFIG_AZURE_IOT_HUB_LOG_LEVEL);
  */
 #define USER_NAME_TEMPLATE	"%s/%s/?api-version=2018-06-30"
 
+#define DPS_USER_NAME		CONFIG_AZURE_IOT_HUB_DPS_ID_SCOPE \
+				"/registrations/%s/api-version=2019-03-31"
+
+/* Topics for data publishing */
+#define TOPIC_TWIN_REPORT	"$iothub/twin/PATCH/properties/reported/?$rid=%d"
+#define TOPIC_EVENTS		"devices/%s/messages/events/%s"
+#define TOPIC_TWIN_REQUEST	"$iothub/twin/GET/?$rid=%d"
+#define TOPIC_DIRECT_METHOD_RES	"$iothub/methods/res/%d/?$rid=%d"
```

Comment:
`request ID` is a string, not just an integer.

Reply:
Is it? I wasn't aware. My (limited) observations until now has been that it's a counter starting from 1 on the first method invocation after a connection is established. I can change to string if it's expected that it can send characters in the ID.

Reply:
[The documentation](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) gives the impression that one could pass anything in the request ID:

> Direct method invocations on a device are HTTPS calls that are made up of the following items:
> ...
> Headers that contain the authorization, request ID, content type, and content encoding.

However it is not documented as a parameter in https://docs.microsoft.com/en-us/rest/api/iothub/service/devices/invokemethod

So I am unsure if it's possible and I need to investigate.

Reply:
When you invoke a direct method from the UI, this is the topic string it arrives on: `$iothub/methods/POST/test/?$rid=1`, so here `rid` is an integer.

When I invoke a direct method with a different name from CLI the rid increases.

```
az iot hub generate-sas-token -n bifravstprodIotHub --du 3600

curl -X POST \
  https://bifravstprodIotHub.azure-devices.net/twins/e66b9f4a-f001-4dcc-8185-4665bffc59fe/methods?api-version=2018-06-30 \
  -H 'Authorization: <sas>' \
  -H 'Content-Type: application/json' \
  -d '{
    "methodName": "reboot",
    "responseTimeoutInSeconds": 200,
    "payload": {
        "input1": "someInput",
        "input2": "anotherInput"
    }
}'
```

```
< $iothub/methods/POST/reboot/?$rid=3
< {"input1":"someInput","input2":"anotherInput"}
```

So, I conclude that `rid` is indeed an integer, and managed by the IoT hub itself.